Apollo/Structures/Tuples.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+
+namespace Apollo.Structures{
+    public struct DoubleTuple {
```

Comment:
Avoid using struct and use class instead. If you must have a structure like this to facilitate the curving algorithm, place it privately inside of Copy instead